models.py

```diff
@@ -48,334 +67,334 @@ def get_indicators(self) -> dict:
+
+        computed_columns = computer.get_computed_columns()
+        indicators = computer.get_indicators()
+        harvest_info = computer.get_harvest_info(
+            [k for k in cls.__dict__.keys() if k.startswith("harvest__")]
+        )
 
-    def get_harvest_info(self, harvest: dict | None) -> HarvestInfo:
-        info = HarvestInfo()
-        if harvest:
-            # No type hints at runtime, we rely on TYPE_MAP being correct
-            for k, v in harvest.items():
-                t = HarvestInfo.__annotations__.get(str(k))
-                if t:
-                    info[k] = Dataset.TYPE_MAP[t](v)
-
-        return info
-
-    def get_license_title(self, id: str | None) -> str | None:
-        return next((item["title"] for item in self.licenses if item["id"] == id), None)
-
-    def to_model(self) -> DatasetRow:
-        return DatasetRow(
-            dataset_id=self.payload["id"],
-            title=self.payload["title"],
-            extras=self.payload["extras"] or {},
-            harvest=self.get_harvest_info(self.payload["harvest"]),
-            last_modified=datetime.fromisoformat(self.payload["last_modified"]),
-            created_at=datetime.fromisoformat(self.payload["created_at"]),
-            slug=self.payload["slug"],
-            acronym=self.payload["acronym"],
-            private=self.payload["private"],
-            spatial=self.payload["spatial"] or {},
-            contact_point=self.payload["contact_point"] or {},
-            organization=self.payload["organization"]["id"]
-            if self.payload["organization"]
-            else None,
-            owner=self.payload["owner"]["id"] if self.payload["owner"] else None,
-            nb_resources=self.payload["resources"]["total"],
-            description=self.payload["description"],
-            frequency=self.payload["frequency"],
-            # tags=self.payload["tags"] or [],
-            temporal_coverage=self.payload["temporal_coverage"] or {},
-            license=self.payload["license"],
-            license__title=self.get_license_title(self.payload["license"]),
-            quality=self.payload["quality"] or {},
-            internal=self.payload["internal"] or {},
-            deleted=False,
+        # conflicts with relationship, needs to be removed after indicators are computed
+        data.pop("resources")
+
+        return cls(
+            **{
+                **{k: v for k, v in data.items() if hasattr(cls, k)},
+                **computed_columns,
+                **indicators,
+                **harvest_info,
+            }
         )
 
-    @classmethod
-    def from_record(cls, record: OrderedDict) -> DatasetRow:
-        return DatasetRow(**{k: v for k, v in record.items() if k != "id"})
 
-    @classmethod
-    def col_types(cls):
+class ResourceComputedColumns:
+    def __init__(self, payload: dict):
+        self.payload = payload
+
+    def get_indicators(self) -> dict:
```

Comment:
I'd be tempted to turn this into a static method of `Resource`. Same for `DatasetComputedColumns`. It's all stateless stuff (but for `DatasetCC` it'll have more impact).

Or at least make `XXXComputedColumns` a subclass of `XXX`, so the two are more clearly tied?

Reply:
Good points about keeping it separate. And no we don't need to handle this now.